staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -244,14 +245,28 @@ func (cfgCtlr *configController) updateObservations() {
 	}
 }
 
+// used from the unit tests only.
+func (cfgCtlr *configController) getPriorityLevelState(plName string) *priorityLevelState {
```

Comment:
Eek, I just noticed, returning this pointer is a concurrency bug: this makes it possible for the caller to read the priorityLevelState without proper synchronization.  Fortunately, the actual caller only cares about whether the pointer is nil or not.  So what we really should have here is ` func (cfgCtlr *configController) hasPriorityLevelState(plName string) bool`.

Reply:
@MikeSpreitzer yes, we should not expose the internal states directly to the caller. This is used by the test only and is used to check it exists. 
I will open a PR with the suggested change on master.

Reply:
it is also worth noting that returning a boolean after releasing the lock means that, in general usage, the boolean may be inaccurate by the time the caller examines it.  Perhaps that should go in a comment on the methond.  Perhaps the method should move into the test file.  In the use case in the test, that fact that the boolean is stale in principle is not a problem in that usage.

Reply:
@MikeSpreitzer  https://github.com/kubernetes/kubernetes/pull/98023